lib/devise/multi_email/models/authenticatable.rb

```diff
@@ -48,22 +48,33 @@ def email=(new_email)
       module ClassMethods
         def find_first_by_auth_conditions(tainted_conditions, opts = {})
           filtered_conditions = devise_parameter_filter.filter(tainted_conditions.dup)
-          email = filtered_conditions.delete(:email)
+          criteria = filtered_conditions.extract!(:email, :unconfirmed_email)
```

Comment:
@allenwq The primary email may not be confirmed upon initial signup. However, Devise doesn't use the `unconfirmed_email` until you intend to *change* the primary email. And I believe that that can't happen until you've confirmed your initial email. So, in effect, the `:unconfirmed_email` hash key only exists in the scenario where Devise is trying to find a user's account based on an email address that is not primary because it is not confirmed yet.

The only way this is not consistent is if there is custom code to make an unconfirmed `Email` record primary. However, in that case, it would search with the `:email` hash key because it's the primary record :-)